Makefile

```diff
@@ -27,8 +27,12 @@ all: bootstrap
 config:
 ifeq (riscv,$(ARCH))
 	@echo $(RISCV_EXEC) > $(OUT)/target
+	@echo \#define TARGET_RISCV 1 > config.h
+	@ln -s riscv-codegen.c $(SRCDIR)/codegen.c
 else
 	@echo $(ARM_EXEC) > $(OUT)/target
+	@echo \#define TARGET_ARM 1 > config.h
+	@ln -s arm-codegen.c $(SRCDIR)/codegen.c
```

Comment:
Ditto.

Reply:
You might need stamp for `config` target.

Reply:
Do you means create a variable for "config" ?

Reply:
No, we should avoid running "make config" twice. Only the first time makes sense. For the second and more times of invoking "make config", there is no effect.

Reply:
Alternatively, you can replace `echo \#define TARGET_RISCV 1 > config.h` with `echo \#define TARGET_RISCV 1 > config` and include `config` in C source. Then, the build system would track the dependency of `config`.

Reply:
Got it.